src/engine/block-utility.js

```diff
@@ -21,6 +21,10 @@ class BlockUtility {
          * @type {?Thread}
          */
         this.thread = thread;
+
+        this._nowObj = {
+            now: () => this.runtime.currentMSecs
```

Comment:
A small performance optional detail: do `this.sequencer.runtime.currentMSecs` instead. Directly dereferencing and avoiding the property call will be faster.

Reply:
@mzgoddard, I'm not sure I understand the change suggested..

Shouldn't `this.runtime` and `this.sequencer.runtime` be the same object?

Reply:
Ah, nvm, just saw the `this.runtime` getter.

Reply:
@mzgoddard, since `this.sequencer` could be null in the constructor, how about the following:

```suggestion
            now: () => this.sequencer && this.sequencer.runtime.currentMSecs
```

Reply:
We don't need worry about being safe with `this.sequencer`. If you call *any* BlockUtility method without setting `sequencer` and `thread` I think you should expect an error. (I want to get rid of the `this.runtime` check here in `nowObj`. I'll open a separate PR for that. Don't need to add more stuff here.)

Anywhere BlockUtility may be used that doesn't set `sequencer` and `thread` during normal vm execution would fair us better to throw an error instead of silently failing.

As well, none of the existing methods check for sequencer or thread.

I guess testing could benefit. But I kind of feel like writing tests that involve blocks without setting up more of the vm they would need to operate and confirm their results would be a disservice to us. Then thrown errors there would also help us.